src/gui/properties/propertieswidget.cpp

```diff
@@ -322,8 +322,14 @@ void PropertiesWidget::loadTorrentInfos(BitTorrent::TorrentHandle *const torrent
 
         // List files in torrent
         m_propListModel->model()->setupModelData(m_torrent->info());
-        if (m_propListModel->model()->rowCount() == 1)
-            m_ui->filesList->setExpanded(m_propListModel->index(0, 0), true);
+        if (m_propListModel->model()->rowCount() == 1) {
+            QModelIndex currentIndex = m_propListModel->index(0, 0);
+            // Expand single-item folders recursively
+            do {
```

Comment:
Could you try refactor the code to omit using `do while` loops? Although it isn't strictly forbidden, yet its usage is very rare in our codebase.